ingestion/src/metadata/profiler/metrics/static/max_length.py

```diff
@@ -58,19 +57,26 @@ def fn(self):
         return None
 
     # pylint: disable=import-outside-toplevel
-    def df_fn(self, df=None):
+    def df_fn(self, dfs=None):
         """dataframe function"""
-        import pandas as pd
         from numpy import vectorize
+        from pandas import notnull
 
-        df = cast(pd.DataFrame, df)  # satisfy mypy
-
+        length_vectorize_func = vectorize(len)
         if self._is_concatenable():
-            length_vector_fn = vectorize(len)
-            return length_vector_fn(
-                df[self.col.name][~df[self.col.name].isnull()]
-            ).max()
+
+            return max(
+                map(
+                    lambda df: length_vectorize_func(
+                        df.astype(object).where(
+                            notnull(df),
+                            "",
+                        )[self.col.name]
+                    ).max(),
+                    dfs,
+                )
+            )
```

Comment:
I am assuming we perform `df.astype(object)` as we'll fall back to a string type correct?

if so we pobably need to do something like:
```python
max(length_vectorize_func(df["a"].fillna("").astype(str)).max() for df in df)
```

Reply:
we were using astype object before to remove the nan using `where(notnull(df), None)`